clients/fides-js/src/lib/consent-banner.tsx

```diff
@@ -0,0 +1,122 @@
+import { h, Component, VNode} from 'preact';
+import { useState } from 'preact/hooks';
+import {CookieKeyConsent, hasSavedConsentCookie, setConsentCookieAcceptAll, setConsentCookieRejectAll,} from "./cookie";
+import {ConsentBannerOptions} from "~/lib/consent-types";
+import debugLog from "~/lib/consent-utils";
+
+
+
+interface BannerProps {
+    defaults: CookieKeyConsent;
+    options: ConsentBannerOptions;
+}
+
+interface BannerState {
+    isShown: boolean
+}
+
+class ConsentBanner extends Component<BannerProps, BannerState> {
+
+    /**
+     * Builds a button DOM element with the given id, class name, and text label
+     */
+    static buildButton = (id: string, className: string, label?: string, onClick?: () => void) => {
+        return(<button id={id} class={`fides-consent-banner-button ${className}`} onClick={onClick ? onClick : undefined}>
+            {label || ""}
+        </button>);
+    }
```

Comment:
Cool - I like seeing some jsx in here 👍 

I'd like us to refactor this further though- it doesn't make much sense to have `buildXYZ`-style functions when you have JSX, we should just have components that we nest.